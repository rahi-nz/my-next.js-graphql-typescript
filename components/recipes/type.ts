export interface Props {
  photo: string;
  title: string;
  id: string;
}
