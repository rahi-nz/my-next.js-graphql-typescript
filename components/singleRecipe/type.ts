import { gql } from "apollo-boost";

export const RECIPE = gql`
  query($id: String!) {
    recipe(id: $id) {
      chef {
        name
      }
      description
      title
      photo {
        url
      }
      tagsCollection {
        items {
          name
        }
      }
    }
  }
`;

export interface Props {
  id: string;
}

interface chef {
  name: string | null;
}

interface photo {
  url: string;
}

interface tag {
  name: string;
}

interface tagsCollection {
  items: Array<tag>;
}

interface recipe {
  chef: chef;
  description: string;
  title: string;
  photo: photo;
  tagsCollection: tagsCollection;
}

export interface getRecipes {
  recipe: recipe;
}
