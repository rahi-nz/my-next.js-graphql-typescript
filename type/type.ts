import { gql } from "apollo-boost";

export const RECIPES = gql`
  query {
    recipeCollection {
      items {
        sys {
          id
        }
        title
        photo {
          url
        }
      }
    }
  }
`;

interface items {
  id: string;
}

interface photo {
  url: string;
}

interface items {
  sys: items;
  title: string;
  photo: photo;
}

interface Collection {
  items: Array<items>;
}

export interface recipesData {
  recipeCollection: Collection;
}
