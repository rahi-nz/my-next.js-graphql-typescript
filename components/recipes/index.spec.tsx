import React from "react";
import renderer from "react-test-renderer";
import Recipes from "./index";
import { Props } from "./type";

describe("<Recipes/>", () => {
  let rootElement;
  let testRenderer;
  const props: Props = {
    photo:
      "https://images.ctfassets.net/kk2bw5ojx476/61XHcqOBFYAYCGsKugoMYK/0009ec560684b37f7f7abadd66680179/SKU1240_hero-374f8cece3c71f5fcdc939039e00fb96.jpg",
    title: "White Cheddar Grilled Cheese with Cherry Preserves & Basil",
    id: "4dT8tcb6ukGSIg2YyuGEOm",
  };
  // eslint-disable-next-line prefer-const,react/display-name
  rootElement = () => <Recipes {...props} />;
  // eslint-disable-next-line prefer-const
  testRenderer = renderer.create(rootElement());

  it("Should render correctly", () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
