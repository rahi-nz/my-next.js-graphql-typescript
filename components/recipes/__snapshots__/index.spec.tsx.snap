// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recipes/> Should render correctly 1`] = `
<div
  className="style__Container-qet6cn-0 bNEakZ"
>
  <a
    href="/recipe/4dT8tcb6ukGSIg2YyuGEOm"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <img
      alt="title"
      className="style__Photo-qet6cn-2 gzrVNn"
      src="https://images.ctfassets.net/kk2bw5ojx476/61XHcqOBFYAYCGsKugoMYK/0009ec560684b37f7f7abadd66680179/SKU1240_hero-374f8cece3c71f5fcdc939039e00fb96.jpg"
    />
    <p
      className="style__Title-qet6cn-1 gFNjlY"
    >
      White Cheddar Grilled Cheese with Cherry Preserves & Basil
    </p>
  </a>
</div>
`;
